frontend/src/libs/components/audio-player/audio-player.tsx

```diff
@@ -1,61 +1,115 @@
-import { useCallback, useRef, useState } from '#libs/hooks/hooks.js';
+import {
+  useAppForm,
+  useCallback,
+  useEffect,
+  useRef,
+} from '#libs/hooks/hooks.js';
 import {
   TRACK_INCREMENT_INDEX,
   TRACK_START_TIME,
 } from '#pages/meditation/libs/constants/constants.js';
 
 import { AudioControls } from './components/audio-controls/audio-controls.js';
 import { ProgressBar } from './components/progress-bar/progress-bar.js';
+import { DEFAULT_AUDIO_OPTIONS } from './libs/constants/constants.js';
+import { AudioOptionKey } from './libs/enums/enums.js';
+import { type AudioControlsHandle } from './libs/types/types.js';
 
-type Properties<T> = {
-  src: string;
+type Properties = {
+  mediaUrl: string;
+  timerDuration: number | null;
   trackIndex: number;
   onSetTrackIndex: (index: number) => void;
-  onSetCurrentTrack: (track: T) => void;
-  tracks: T[];
+  tracksCount: number;
 };
 
-const AudioPlayer = <T,>({
-  src,
+const AudioPlayer = ({
+  mediaUrl,
+  timerDuration,
   trackIndex,
-  tracks,
-  onSetCurrentTrack,
+  tracksCount,
   onSetTrackIndex,
-}: Properties<T>): JSX.Element => {
-  const [timeProgress, setTimeProgress] = useState(TRACK_START_TIME);
-  const [duration, setDuration] = useState(TRACK_START_TIME);
-
+}: Properties): JSX.Element => {
```

Comment:
```suggestion
})=> {
```
We can use `React.FC<Properties>`